samples/bluetooth/encrypted_advertising/peripheral/src/peripheral_ead.c

```diff
@@ -0,0 +1,377 @@
+
+	err = bt_le_ext_adv_start(adv, &start_params);
+	if (err) {
+		LOG_ERR("Failed to start advertiser (%d)", err);
+		return -1;
+	}
+
+	LOG_DBG("Advertiser started");
+
+	return 0;
+}
+
+static int stop_and_delete_adv(struct bt_le_ext_adv *adv)
+{
+	int err;
+
+	err = bt_le_ext_adv_stop(adv);
+	if (err) {
+		LOG_ERR("Failed to stop advertiser (err %d)", err);
+		return -1;
+	}
+
+	err = bt_le_ext_adv_delete(adv);
+	if (err) {
+		LOG_ERR("Failed to delete advertiser (err %d)", err);
+		return -1;
+	}
+
+	return 0;
+}
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (err) {
+		LOG_ERR("Failed to connect to %s (%u)", addr, err);
+		return;
+	}
+
+	LOG_DBG("Connected to %s", addr);
+
+	default_conn = conn;
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	LOG_DBG("Disconnected from %s (reason 0x%02x)", addr, reason);
+
+	k_poll_signal_raise(&disconn_signal, 0);
+}
+
+static void security_changed(struct bt_conn *conn, bt_security_t level, enum bt_security_err err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (!err) {
+		LOG_DBG("Security changed: %s level %u", addr, level);
+	} else {
+		LOG_DBG("Security failed: %s level %u err %d", addr, level, err);
+	}
+}
+
+static void auth_passkey_confirm(struct bt_conn *conn, unsigned int passkey)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+	char passkey_str[7];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	snprintk(passkey_str, 7, "%06u", passkey);
+
+	LOG_DBG("Confirm passkey for %s: %s", addr, passkey_str);
+
+	k_poll_signal_raise(&passkey_enter_signal, 0);
+}
+
+static void auth_passkey_display(struct bt_conn *conn, unsigned int passkey)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+	char passkey_str[7];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	snprintk(passkey_str, 7, "%06u", passkey);
+
+	LOG_DBG("Passkey for %s: %s", addr, passkey_str);
+}
+
+static void auth_cancel(struct bt_conn *conn)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	LOG_DBG("Pairing cancelled: %s", addr);
+}
+
+static int update_security(void)
+{
+	int err;
+
+	await_signal(&passkey_enter_signal);
+
+	err = bt_conn_auth_passkey_confirm(default_conn);
+	if (err) {
+		LOG_DBG("Failed to confirm passkey.");
+		return -1;
+	}
+
+	return 0;
+}
+
+static int init_bt(void)
+{
+	int err;
+
+	k_poll_signal_init(&disconn_signal);
+	k_poll_signal_init(&passkey_enter_signal);
+
+	err = bt_enable(NULL);
+	if (err) {
+		LOG_ERR("Bluetooth init failed (err %d)", err);
+		return -1;
+	}
+
+	LOG_DBG("Bluetooth initialized");
+
+	err = bt_unpair(BT_ID_DEFAULT, BT_ADDR_LE_ANY);
+	if (err) {
+		LOG_ERR("Unpairing failed (err %d)", err);
+	}
```

Comment:
Instead of doing this, why not just disabling support for bonding? Same result, but less effort

Reply:
No, we want the central to resolve the RPA which it does not with `CONFIG_BT_BONDABLE=n`.

Reply:
Aer you saying that the feature depends on having a bond? If that is the case, then `BT_EAD` should have a dependency on `BT_BONDABLE`. 

If it does not depend on having a bond, why do we need to solve the RPA?

Reply:
Looking at the prj.conf file, it doesn't look like you enable `BT_BONDABLE` in any case, so it doesn't look like you have any bonds anyways, and then this calls doesn't really do anything, right?

Reply:
> Aer you saying that the feature depends on having a bond? If that is the case, then `BT_EAD` should have a dependency on `BT_BONDABLE`.
>
> If it does not depend on having a bond, why do we need to resolve the RPA?

No, `BT_EAD` doesn't depend on `BT_BONDABLE`. I use the resolved RPA to get the adv data of the peer I exchanged the Key Material with.

> Looking at the prj.conf file, it doesn't look like you enable `BT_BONDABLE` in any case, so it doesn't look like you have any bonds anyways, and then this calls doesn't really do anything, right?

`BT_BONDABLE` default to `y`.